libraries/engage/core/hocs/withNavigation.jsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { push, pop, replace, reset } from '../router/helpers';
+
+/**
+ * Injects navigation functions into the desired component.
+ * @param {Function} WrappedComponent The react component to wrap.
+ * @returns {JSX}
+ */
+export function withNavigation(WrappedComponent) {
+  return props => (
+    <WrappedComponent
+      {...props}
+      push={push}
```

Comment:
I wonder if make sense to prefix the props with e.g. `history`. When one injects them via the hook, they are already kind of scoped due to the way, hooks work. But when i think about the usage with the HOC, they will be mixed with the other props of the component, which might be confusing.

Reply:
It does make sense. I changed it.